### DISCUSSION TITLE
Ansible playbook gets connection refused

### DISCUSSION THREAD

**Discussion by alevikpes** [UtrechtUniversity/yoda#480] [Q&A] [answered]
Hello!

In the step 4 of the guide the second command `ansible-playbook -i environments/development/allinone/ playbook.yml` gives me the following:
```bash
TASK [opensearch : Discover presence of index] ***********************************************************************************************************************************************************************************************************************************
FAILED - RETRYING: [combined]: Discover presence of index (10 retries left).
FAILED - RETRYING: [combined]: Discover presence of index (9 retries left).
FAILED - RETRYING: [combined]: Discover presence of index (8 retries left).
FAILED - RETRYING: [combined]: Discover presence of index (7 retries left).
FAILED - RETRYING: [combined]: Discover presence of index (6 retries left).
FAILED - RETRYING: [combined]: Discover presence of index (5 retries left).
FAILED - RETRYING: [combined]: Discover presence of index (4 retries left).
FAILED - RETRYING: [combined]: Discover presence of index (3 retries left).
FAILED - RETRYING: [combined]: Discover presence of index (2 retries left).
FAILED - RETRYING: [combined]: Discover presence of index (1 retries left).
fatal: [combined]: FAILED! => {"attempts": 10, "changed": false, "elapsed": 0, "msg": "Status code was -1 and not [200, 404]: Request failed: <urlopen error [Errno 111] Connection refused>", "redirected": false, "status": -1, "url": "http://combined.yoda.test:9200/yoda"}
```

Is there anything to be exposed on the VM?

**Comment by stsnel** [comment]
Hello,

I would normally expect this step to complete automatically, without any intervention from the person running the playbook. 

Perhaps something is going wrong with name resolution or the OpenSearch service itself. Can you first check that OpenSearch is running normally?

```
sudo systemctl status opensearch
sudo journalctl -u opensearch
```

**Reply by alevikpes** [reply]
I don't have opensearch on my os. Or is it in VM?

**Reply by lwesterhof** [reply]
OpenSearch should be installed on the VM

**Reply by alevikpes** [reply]
The opensearch looks good:
```bash
vagrant@combined:~$ sudo systemctl status opensearch
● opensearch.service - opensearch
     Loaded: loaded (/etc/systemd/system/opensearch.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2024-10-10 12:20:46 CEST; 3h 31min ago
   Main PID: 20309 (java)
      Tasks: 32 (limit: 4595)
     Memory: 1.2G
     CGroup: /system.slice/opensearch.service
             └─20309 /usr/share/opensearch/jdk/bin/java -Xshare:auto -Dopensearch.networkaddress.cache.ttl=60 -Dopensearch.networkaddress.cache.negative.ttl=10 -XX:+AlwaysPreTouch -Xss1m -Djava.awt.headless=true -Dfile.encoding=UTF-8 -Djna.nosys=true -XX:-OmitStackTraceIn>

Oct 10 12:20:46 combined systemd[1]: Started opensearch.

vagrant@combined:~$ sudo journalctl -u opensearch
-- Logs begin at Thu 2024-10-10 12:03:00 CEST, end at Thu 2024-10-10 15:52:22 CEST. --
Oct 10 12:20:46 combined systemd[1]: Started opensearch.
```

**Reply by stsnel** [reply, marked as answer]
Right, it looks like OpenSearch is running fine based on this output.

Do you get a JSON response now if you connect to OpenSearch (e.g. `curl http://combined.yoda.test:9200/yoda` from the VM)?

If so, can you try re-running the playbook to see if it still fails?

**Reply by alevikpes** [reply]
yes, I am getting a 404:

> {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [yoda]","index":"yoda","resource.id":"yoda","resource.type":"index_or_alias","index_uuid":"_na_"}],"type":"index_not_found_exception","reason":"no such index [yoda]","index":"yoda","resource.id":"yoda","resource.type":"index_or_alias","index_uuid":"_na_"},"status":404}

rerunning the playbook ...

**Reply by alevikpes** [reply]
Rerunning fixed the previous problem but it got another one:

> TASK [irods_runtime : Check if systemd service is active for handler] ***********************************************************************************************************************************************************************************************************
fatal: [combined]: FAILED! => {"changed": false, "cmd": ["systemctl", "is-active", "irods"], "delta": "0:00:00.025521", "end": "2024-10-10 16:32:52.444575", "msg": "non-zero return code", "rc": 3, "start": "2024-10-10 16:32:52.419054", "stderr": "", "stderr_lines": [], "stdout": "inactive", "stdout_lines": ["inactive"]}
...ignoring

It repeated 2 times and stopped with:

> TASK [yoda_davrods : Download DavRODS package] **********************************************************************************************************************************************************************************************************************************
fatal: [combined]: FAILED! => {"changed": false, "dest": "/tmp/davrods-4.2.12-1.5.1.deb", "elapsed": 20, "msg": "Request failed: <urlopen error [Errno -3] Temporary failure in name resolution>", "url": "https://github.com/UtrechtUniversity/davrods/releases/download/4.2.12_1.5.1/davrods-4.2.12-1.5.1.deb"}

The url is fine

**Reply by alevikpes** [reply]
```bash
vagrant@combined:~$ sudo systemctl status irods
Unit irods.service could not be found.
```
```bash
vagrant@combined:~$ sudo systemctl is-active irods
inactive
```

**Reply by alevikpes** [reply]
I re-ran the playbook and something was fixed. I can see the service is available and running.

There were some other errors but the playbook finished normally.

**Reply by stsnel** [reply]
Good to hear that the playbook finished eventually :+1:  These errors should not happen normally, but we've observed them before in situations where DNS name resolution was a bit flaky.

The workaround we use in these situations is to configure the VM to use Google's DNS servers, which seem to be reliable. You can do this by adding a setting `common_custom_dns_enable: true` to `environments/development/allinone/group_vars/allinone.yml` before running the playbook.

**Reply by alevikpes** [reply]
Thank you, will try that.

**Comment by alevikpes** [comment]
btw, what is the credentials for the ubuntu in VM? Can't find it anywhere

**Reply by stsnel** [reply]
You should be able to log in on the VM using `vagrant ssh` (the credentials are handled by Vagrant).

**Reply by alevikpes** [reply]
nice, thank you
